app/assets/javascripts/discourse/app/components/post/menu.gjs

```diff
@@ -609,6 +620,7 @@ export default class PostMenu extends Component {
   }
 
   <template>
+    {{this.checkLikedPost}}
```

Comment:
@davidtaylorhq / @CvX / @megothss - I'm not 100% sure about _how_ the check is done. This will be called on every render of a post menu. Maybe there's a better way?

Reply:
Sounds like the lowest-friction thing is to keep it then 😅 

If we can refactor it so that it's called from the constructor instead of a side-effect-getter-from-a-template, then I'd be ok with it. We should also see what @megothss thinks though.

Reply:
I _might_ be able to send the "I liked post 123" when signing up / logging in so the server can like the post directly on the server once the user is properly authed, but that might be a little bit more work to get right. It would solve the "check on every post" penalty though.

Happy to hear if @megothss has a better idea 👍